apps/condo/domains/acquiring/access/Payment.js

```diff
@@ -26,29 +28,40 @@ async function canReadPayments ({ authentication: { item: user }, listKey }) {
         return { multiPayment: { user: { id: user.id } } }
     }
 
+    const filters = [
+        // Acquiring integration account can see it's payments
+        { context: { integration: { accessRights_some: { user: { id: user.id }, deletedAt: null } } } },
+        // Employee with `canReadPayments` can see theirs organization payments
+        {
+            AND: [
+                {
+                    invoice_is_null: true,
+                    organization: { employees_some: { user: { id: user.id }, role: { canReadPayments: true }, deletedAt: null, isBlocked: false } },
+                },
+            ],
+        },
+        // Employee with `canReadPaymentsWithInvoices` can see theirs organization payments with invoices
+        {
+            AND: [
+                {
+                    invoice_is_null: false,
+                    organization: { employees_some: { user: { id: user.id }, role: { canReadPaymentsWithInvoices: true }, deletedAt: null, isBlocked: false } },
+                },
+            ],
+        },
+    ]
+
+    if (user.type === SERVICE) {
+        const filtersByOrganization = await canReadObjectsAsB2BAppServiceUser(args)
+        if (filtersByOrganization !== false) {
+            filters.push({ AND: [{ ...filtersByOrganization }] })
+        }
+    }
+
+    console.error(filters)
+
     return {
-        OR: [
-            // Acquiring integration account can see it's payments
-            { context: { integration: { accessRights_some: { user: { id: user.id }, deletedAt: null } } } },
-            // Employee with `canReadPayments` can see theirs organization payments
-            {
-                AND: [
-                    {
-                        invoice_is_null: true,
-                        organization: { employees_some: { user: { id: user.id }, role: { canReadPayments: true }, deletedAt: null, isBlocked: false } },
-                    },
-                ],
-            },
-            // Employee with `canReadPaymentsWithInvoices` can see theirs organization payments with invoices
-            {
-                AND: [
-                    {
-                        invoice_is_null: false,
-                        organization: { employees_some: { user: { id: user.id }, role: { canReadPaymentsWithInvoices: true }, deletedAt: null, isBlocked: false } },
-                    },
-                ],
-            },
-        ],
+        OR: filters,
```

Comment:
i think it's better to separate these filters for different user types, so that the query is not so complex.  like so:
if (user.type === SERVICE) {
      // here logic only for service users
 }

if (user.type === STAFF) {
    // logic for STAFF users
}

return false